pytest_asyncio/plugin.py

```diff
@@ -602,6 +616,62 @@ def _set_event_loop(loop: AbstractEventLoop | None) -> None:
         asyncio.set_event_loop(loop)
 
 
+_session_loop: contextvars.ContextVar[asyncio.AbstractEventLoop | None] = (
+    contextvars.ContextVar(
+        "_session_loop",
+        default=None,
+    )
+)
+_package_loop: contextvars.ContextVar[asyncio.AbstractEventLoop | None] = (
+    contextvars.ContextVar(
+        "_package_loop",
+        default=None,
+    )
+)
+_module_loop: contextvars.ContextVar[asyncio.AbstractEventLoop | None] = (
+    contextvars.ContextVar(
+        "_module_loop",
+        default=None,
+    )
+)
+_class_loop: contextvars.ContextVar[asyncio.AbstractEventLoop | None] = (
+    contextvars.ContextVar(
+        "_class_loop",
+        default=None,
+    )
+)
+_function_loop: contextvars.ContextVar[asyncio.AbstractEventLoop | None] = (
+    contextvars.ContextVar(
+        "_function_loop",
+        default=None,
+    )
+)
+
+_SCOPE_TO_CONTEXTVAR = {
```

Comment:
The current way of handling contextvars with copying them and restoring them was introduced before switching to *asyncio.Runner*, if I remember correctly.

My understanding is that *asyncio.Runner* holds a default context along with its loop. We already have one Runner per loop scope. Do you think there's any way we can utilize the Runner's default context rather than storing contexts per scope externally?

This is just something to think about. If it's not possible or it would take a lot more work, we can start with this solution as well.